applications/ecallmgr/src/ecallmgr_call_events.erl

```diff
@@ -462,10 +466,32 @@ handle_info('sanity_check', #state{callid=CallId}=State) ->
     end;
 handle_info('shutdown', State) ->
     {'stop', 'normal', State};
+handle_info(?LOOPBACK_BOWOUT_MSG(Node, Props), #state{call_id=ResigningUUID
+                                                      ,node=Node
+                                                     }=State) ->
+    NewUUID = handle_bowout(Node, Props, ResigningUUID),
+    {'noreply', State#state{call_id=NewUUID}};
 handle_info(_Info, State) ->
     lager:debug("unhandled message: ~p", [_Info]),
     {'noreply', State}.
 
+-spec handle_bowout(atom(), wh_proplist(), ne_binary()) -> ne_binary().
+handle_bowout(Node, Props, ResigningUUID) ->
+    AcquiringUUID = props:get_value(?ACQUIRED_UUID, Props),
+    case props:get_value(?RESIGNING_UUID, Props) of
```

Comment:
The bowout will happen on call forwarding and some other use-cases where the call-id may not actually change.  We shouldn't unregister then re-register for events for the same call-id ;)